src/BuiltInTools/DotNetDeltaApplier/Microsoft.Extensions.DotNetDeltaApplier.csproj

```diff
@@ -1,7 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <!-- Intentionally pinned. This feature is supported in projects targeting 6.0 or newer.-->
-    <TargetFramework>net6.0</TargetFramework>
```

Comment:
My biggest concern is here: I couldn't find a good reason why this says its pinned. Is there any extra validation I can do to make sure this doesn't break anything?

If absolutely necessary, we can set the TFM to `net7.0` for source-build and `net6.0;net7.0` otherwise.